### PR TITLE
Pass locally defined scopes to RemoteClientConfigStore

### PR DESCRIPTION
**Pull request by franco-bocci**
# TL;DR
This PR fixes a bug in the auth mechanism where locally defined `scopes` are ignored and the ones defined in the admin are always used.

## Type
 - [x] Bug Fix
 - [ ] Feature
 - [ ] Plugin

## Are all requirements met?

 - [x] Code completed
 - [ ] Smoke tested
 - [x] Unit tests added
 - [x] Code documentation added
 - [ ] Any pending items have an associated Issue

## Complete description
If `scopes` are locally defined, use those for auth. This mirrors the reference implementation [here](https://github.com/flyteorg/flyteidl/blob/e1a667ef2536bbfc61331490b5417404f45ddc0b/clients/go/admin/token_source_provider.go#L67).

## Tracking Issue
https://github.com/flyteorg/flyte/issues/3486

### CODE REVIEW

File: flytekit/clients/auth_helper.py
Comment (kumare3) [inline flytekit/clients/auth_helper.py:42]: i think we should not mess around with this, but add scopes to the constructor of individual authenticators. this way it remains clear

Comment (franco-bocci) [reply flytekit/clients/auth_helper.py:42]: mmm, okay. Would you add it for all authenticators? Or just for `ClientCredentialsAuthenticator`?

Comment (wild-endeavor) [reply flytekit/clients/auth_helper.py:42]: let's just do the clientcredentials one for now please.

Comment (franco-bocci) [reply flytekit/clients/auth_helper.py]: Cool. Done!

**Comment by franco-bocci** [comment]
I removed the other commit from the commit history as the first approach was not correct as discussed. Let me know what you think!

File: flytekit/clients/auth/authenticator.py
Comment (kumare3) [inline flytekit/clients/auth/authenticator.py:167]:
```suggestion
        scopes: typing.Optional[typing.List[str]] = None,
```

File: flytekit/clients/auth/authenticator.py
Comment (kumare3) [inline flytekit/clients/auth/authenticator.py:166]: also for this - sorry
```
str -> typing.Optional[str]
```

**Comment by kumare3** [comment]
@franco-bocci thank you - after this change +1

**Review by wild-endeavor — APPROVED** [review]

**Comment by franco-bocci** [comment]
BTW, I can't merge this as I don't have write access. How is the merge + release handled? Of course, redirect me to the corresponding docs if I missed that part 😄

**Pull request merged**
